### ISSUE TITLE
Add test supporting TextPlacement

### ISSUE THREAD

**Issue by DavidGrant-NIWC** [iho-ohi/S-164-Sub-Group#69]
* Placement of feature name (`textType` = `1`)
* Placement of light description (`textType` = `2`)
* Placement of text annotation (`text` = "Arbitrary label")

At least one of these should have `textRotation` = `true`.
At least one should have `scaleMinimum` populated (probably the text annotation).

**Comment by kusala9** [comment]
Suggest these tests are added into 3.8.5 as more TextDisplay tests. The Priorities datasets can accomodate the test cases. Does the current PC (1.1 or 1.2?) support these, if so I'll draft the dataset contents to look at.

**Comment by DavidGrant-NIWC** [comment]
> Does the current PC (1.1 or 1.2?) support these

No, I need the data to test/validate an implementation...

**Comment by kusala9** [comment]
ok, I can make a simple cell up, then let me know what feature changes you want.

**Comment by kusala9** [comment]
the QobD cell I put together has a point feature with TextPlacement in it. Is this ok? I can add more features with the test cases above if that works? cell copied below - so we can close the QobD Issue.

[32.zip](https://github.com/iho-ohi/S-164-Sub-Group/files/14212399/32.zip)

**Comment by DavidGrant-NIWC** [comment]
Good enough to start testing, but note:

![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/04750f5d-b087-487f-b32a-26568daee137)

Also, the SCAMIN values are a little odd:
`DataCoverage`: 45k-180k
`LightAllAround`: 500k (feature associated with `TextPlacement`)
`CardinalBeacon`: 150k
`TextPlacement`: 150k

**Comment by kusala9** [comment]
agreed, I'll adjust the SCAMIN (they're hangovers from the original cell). I will do a few more examples with appropriate encoding in that cell and upload for testing.

**Comment by DavidGrant-NIWC** [comment]
Looks good (note: we haven't implemented `textRotation` yet).

![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/a90e989a-ad2b-4215-b22d-d44cef58c8ad)

**Comment by DavidGrant-NIWC** [comment]
Would you mind adding a sounding with name encoded?

https://github.com/iho-ohi/S-101-Documentation-and-FC/issues/124#issuecomment-1952405594
https://github.com/iho-ohi/S-101_Portrayal-Catalogue/issues/339

**Comment by kusala9** [comment]
the 42m sounding in this one has a feature name defined. I'll sort out the better textPlacement features later.

[32v3.zip](https://github.com/iho-ohi/S-164-Sub-Group/files/14344302/32v3.zip)

**Comment by DavidGrant-NIWC** [comment]
With `textRotation`:

|North Up|Rotated -45|
|:---:|:---:|
|![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/789575d9-6184-4886-86e7-38cc70ef2f3f)|![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/f7b8ba9f-f84a-4643-aca5-1acabd05585a)

**Comment by kusala9** [comment]
v5 (edition 9) of this cell, now has a few more textPlacement examples. Various offset/bearings. Also, one with SCAMIN. 
light characrteristic doesn't seem to work on textplacement. there's an identical light without TP.

[32v5.zip](https://github.com/iho-ohi/S-164-Sub-Group/files/14370808/32v5.zip)

**Comment by DavidGrant-NIWC** [comment]
> light characrteristic doesn't seem to work on textplacement.

See https://github.com/iho-ohi/S-101-Documentation-and-FC/issues/129

|Rotation|Result|
|---:|:---:|
|0|![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/6a56b3f8-91d0-4f76-8834-94089c9a3a6b)
|90|![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/c05ba820-1e0c-47b6-acee-9738b66afb36)
|180|![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/11173eca-cf50-4051-83ec-ef9b7a120ef6)
|270|![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/fc8c86d5-5d89-4e90-9cf2-a09b48ae6c2e)

**Comment by DavidGrant-NIWC** [comment]
> If the FC association binding is changed to 0..2 then does the light characteristic work? My copy of the viewer (with latest PC) doesn't show it still....

Sorry, I misunderstood. You need an updated testbed; the text is rotated using GeographicCRS and we hadn't implemented that for TextInstruction until very recently. We should be sending Yong an updated release package today or tomorrow; it should be available from the registry.

![image](https://github.com/iho-ohi/S-164-Sub-Group/assets/62906211/245d785a-a597-43be-bd31-2ef90ac649b6)
